Index: linux-6.12.65/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
===================================================================
--- linux-6.12.65.orig/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
+++ linux-6.12.65/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
@@ -420,6 +420,7 @@ stmmac_probe_config_dt(struct platform_d
 	struct plat_stmmacenet_data *plat;
 	struct stmmac_dma_cfg *dma_cfg;
 	static int bus_id = -ENODEV;
+	const char *label;
 	int phy_mode;
 	void *ret;
 	int rc;
@@ -467,6 +468,11 @@ stmmac_probe_config_dt(struct platform_d
 	/* Default to phy auto-detection */
 	plat->phy_addr = -1;
 
+	/* Get custom label for network device name */
+	if (!of_property_read_string(np, "label", &label)) {
+		plat->label = devm_kstrdup(&pdev->dev, label, GFP_KERNEL);
+	}
+
 	/* Default to get clk_csr from stmmac_clk_csr_set(),
 	 * or get clk_csr from device tree.
 	 */
@@ -729,7 +735,6 @@ int stmmac_get_platform_resources(struct
 	if (stmmac_res->wol_irq < 0) {
 		if (stmmac_res->wol_irq == -EPROBE_DEFER)
 			return -EPROBE_DEFER;
-		dev_info(&pdev->dev, "IRQ eth_wake_irq not found\n");
 		stmmac_res->wol_irq = stmmac_res->irq;
 	}
 
@@ -738,7 +743,6 @@ int stmmac_get_platform_resources(struct
 	if (stmmac_res->lpi_irq < 0) {
 		if (stmmac_res->lpi_irq == -EPROBE_DEFER)
 			return -EPROBE_DEFER;
-		dev_info(&pdev->dev, "IRQ eth_lpi not found\n");
 	}
 
 	stmmac_res->sfty_irq =
@@ -746,7 +750,6 @@ int stmmac_get_platform_resources(struct
 	if (stmmac_res->sfty_irq < 0) {
 		if (stmmac_res->sfty_irq == -EPROBE_DEFER)
 			return -EPROBE_DEFER;
-		dev_info(&pdev->dev, "IRQ sfty not found\n");
 	}
 
 	stmmac_res->addr = devm_platform_ioremap_resource(pdev, 0);
Index: linux-6.12.65/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
===================================================================
--- linux-6.12.65.orig/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ linux-6.12.65/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -7806,6 +7806,10 @@ int stmmac_dvr_probe(struct device *devi
 		goto error_phy_setup;
 	}
 
+	if (plat_dat->label) {
+		strscpy(ndev->name, plat_dat->label, IFNAMSIZ);
+	}
+
 	ret = register_netdev(ndev);
 	if (ret) {
 		dev_err(priv->device, "%s: ERROR %i registering the device\n",
Index: linux-6.12.65/include/linux/stmmac.h
===================================================================
--- linux-6.12.65.orig/include/linux/stmmac.h
+++ linux-6.12.65/include/linux/stmmac.h
@@ -220,6 +220,7 @@ struct plat_stmmacenet_data {
 	int maxmtu;
 	int multicast_filter_bins;
 	int unicast_filter_entries;
+	const char *label;
 	int tx_fifo_size;
 	int rx_fifo_size;
 	u32 host_dma_width;
